index.bs

```diff
@@ -3739,49 +4068,260 @@ partial dictionary MLOpSupportLimits {
     :: {{MLSupportLimits}} for output operand.
 </dl>
 
-{{MLOpSupportLimits}} has the following members for {{MLGraphBuilder/gather()}}:
+{{MLOpSupportLimits}} has the following members for {{MLGraphBuilder/gather()}}:
+<dl dfn-type=dict-member dfn-for=MLOpSupportLimits>
+    : <dfn>gather</dfn>
+    :: Support limits for operator {{MLGraphBuilder/gather()}}.
+</dl>
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>gather(|input|, |indices|, |options|)</dfn> method steps are:
+  </summary>
+    1. If [=this=] [=MLGraphBuilder/can not build=], then [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
+    1. If [=MLGraphBuilder/validating operand=] with [=this=] and any of |input| and |indices| returns false, then [=exception/throw=] a {{TypeError}}.
+    1. If |indices|'s [=MLOperand/dataType=] is not one of its [=/allowed data types=] (according to [this table](#constraints-gather)), then [=exception/throw=] a {{TypeError}}.
+    1. Let |shapeInput| be |input|'s [=MLOperand/shape=] and |rankInput| be |shapeInput|'s [=MLOperand/rank=].
+    1. Let |shapeIndices| be |indices|'s [=MLOperand/shape=].
+    1. Let |axis| be |options|.{{MLGatherOptions/axis}}.
+    1. If |axis| is greater than or equal to |rankInput|, then [=exception/throw=] a {{TypeError}}.
+    1. Let |dimCount| be zero.
+    1. Let |rankOutput| be zero.
+    1. Let |shapeOutput| be an empty list.
+    1. [=list/For each=] |size| of |shapeInput|:
+        1. If |dimCount| is equal to |axis| then [=iteration/break=].
+        1. Set |shapeOutput|[|dimCount|] to |size|.
+        1. Increment |dimCount| by one.
+    1. Set |rankOutput| to |dimCount|.
+    1. Let |dimCount| be zero.
+    1. [=list/For each=] |size| of |shapeIndices|:
+        1. Set |shapeOutput|[|rankOutput| + |dimCount|] to |size|.
+        1. Increment |dimCount| by one.
+    1. Set |rankOutput| to |rankOutput| + |dimCount|.
+    1. Let |dimCount| be zero.
+    1. [=list/For each=] |size| of |shapeInput|:
+        1. If |dimCount| is less than or equal to |axis| then [=iteration/continue=].
+        1. Set |shapeOutput|[|rankOutput| + |dimCount| - |axis| - 1] to |size|.
+        1. Increment |dimCount| by one.
```

Comment:
This `dimCount` isn't used after it is calculated?

(Pre-existing issue!)

Reply:
Is it not used in the `for` loop to assign size?